File: 102_Binary_Tree_Level_Order_Traversal/xsy/solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func levelOrder(root *TreeNode) [][]int {
	result := l(root, 0,  [][]int{})
	return result
}

func l(root *TreeNode, level int, box [][]int) [][]int {
	if root == nil {
		return box
	}
	if (len(box) < level + 1) {
		box = append(box, []int{})
	}
	box[level] = append(box[level], root.Val)
	box = l(root.Left, level+1, box)
	box = l(root.Right, level+1, box)
	return box
}

File: 102_Binary_Tree_Level_Order_Traversal/xsy/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        r = []
        def _levelOrder(root, level):
            if root is None:
                return
            if len(r) < level + 1:
                r.append([])
                r[level].append(root.val)
            _levelOrder(root.left, level + 1)
            _levelOrder(root.right, level + 1)
        _levelOrder(root, 0)
        return r
